pilot/pkg/model/gateway.go

```diff
@@ -167,6 +168,13 @@ func MergeGateways(gateways []gatewayWithInstances) *MergedGateway {
 					if s.Tls.Mode == networking.ServerTLSSettings_AUTO_PASSTHROUGH {
 						autoPassthrough = true
 					}
+					if len(s.Tls.CipherSuites) > 0 {
+						serverIdentifier := s.Name
+						if len(serverIdentifier) == 0 {
+							serverIdentifier = s.Port.Name
+						}
+						s.Tls.CipherSuites = filteredCipherSuites(s.Tls.CipherSuites, serverIdentifier, gatewayName)
```

Comment:
is it safe to mutate here? Will this impact the underlying client-go cache?

Reply:
I didnot realize it is mutating the original object

Reply:
So is it OK if filteredCipherSuite returns a different set, we clone the server and set?

Reply:
What is this pr fixing?

Reply:
#34109 added validation and logic to skip invalid cipher suites. But the logic to skip is executed on every listener creation. I am trying to see if we can do it once with this PR.